File: PS/sheet3/vasya_and_string.cs
using System;

class Program
{
    static int Solve(string s, int k, char target)
    {
        int l = 0, cnt = 0, best = 0;

        for (int r = 0; r < s.Length; r++)
        {
            if (s[r] != target)
                cnt++;

            while (cnt > k)
            {
                if (s[l] != target)
                    cnt--;
                l++;
            }

            best = Math.Max(best, r - l + 1);
        }

        return best;
    }

    static void Main()
    {
        var first = Console.ReadLine().Split();
        int n = int.Parse(first[0]);
        int k = int.Parse(first[1]);

        string s = Console.ReadLine();

        int ans = Math.Max(Solve(s, k, 'a'), Solve(s, k, 'b'));
        Console.WriteLine(ans);
    }
}

File: Problem Solving/sheet3/vasya_and_string.cpp
#include <bits/stdc++.h>
using namespace std;

int solve(const string& s, int k, char target) {
    int l = 0, count = 0, best = 0;

    for (int r = 0; r < (int)s.size(); r++) {
        if (s[r] != target)
            count++;

        while (count > k) {
            if (s[l] != target)
                count--;
            l++;
        }

        best = max(best, r - l + 1);
    }

    return best;
}

int main() {
    int n, k;
    cin >> n >> k;

    string s;
    cin >> s;

    int ans = max(solve(s, k, 'a'), solve(s, k, 'b'));

    cout << ans << endl;
    return 0;
}